### PR TITLE
[New Term Entry] Adding compareTo() to Java date terms

### LINKED ISSUE #2836: [Term Entry] Java Date compareTo
### Reason/inspiration (optional)

We would like a new term entry under Java [Date](https://www.codecademy.com/resources/docs/java/date) for the Java Date `.compareTo()` method. 
Provide the entry in a markdown file under `docs/content/java/concepts/date/terms/compareTo/compareTo.md`

The entry should
    - Have an introductory paragraph about what the method is and does.
    - A syntax section with a pseudo code block with an explanation of it.
    - An example showing the method in use.

Please refer to the [concept entry templates](https://github.com/Codecademy/docs/blob/main/documentation/concept-entry-template.md), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by ChiragAgg5k**
### Description

Adds a new term entry for compareTo() - a Date object comparison method - to the Java dates section. Closes #2836 .

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@ChiragAgg5k I did a first review and only had a couple of comments.

**Comment by ChiragAgg5k** [comment]
Made the changes as requested 👍🏻

**Review by SSwiniarski — APPROVED** [review]
@ChiragAgg5k Looks good for a second review.

File: content/java/concepts/date/terms/compareTo/compareTo.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+Tags:
+  - 'Date'
+  - 'Methods'
+  - 'Comparison'
+CatalogContent:
+  - 'learn-java'
+  - 'paths/computer-science'
+---
+
+The **`Date.compareTo()`** method compares one Date object with another and returns an integer value accordingly.
```
Comment (caupolicandiaz) [inline content/java/concepts/date/terms/compareTo/compareTo.md:16]:
```suggestion
The **`Date.compareTo()`** method compares one `Date` object with another and returns an integer value accordingly.
```

File: content/java/concepts/date/terms/compareTo/compareTo.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+```pseudo
+thisDate.compareTo(argumentDate)
+```
+
+The `Date.compareTo()` takes only one parameter:
+
+- `argumentDate`: A `Date` object to be compared chronologically with the given date.
+
+It returns
```
Comment (caupolicandiaz) [inline content/java/concepts/date/terms/compareTo/compareTo.md:28]:
```suggestion
It returns:
```

File: content/java/concepts/date/terms/compareTo/compareTo.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+- the value `0` if the argument Date is equal to this Date.
```
Comment (caupolicandiaz) [inline content/java/concepts/date/terms/compareTo/compareTo.md:30]:
```suggestion
- The value `0` if `argumentDate` is equal to `thisDate`.
```

Comment (caupolicandiaz) [reply content/java/concepts/date/terms/compareTo/compareTo.md:30]: Update the other bullet points with the formatting shown above

File: content/java/concepts/date/terms/compareTo/compareTo.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+- a value less than `0` if this Date is before the Date argument.
+- a value greater than `0` if this Date is after the Date argument.
+
+The function throws a single exception that is [NullPointerException](https://www.codecademy.com/resources/docs/java/errors/nullpointerexception) if the argumentDate is null.
```
Comment (caupolicandiaz) [inline content/java/concepts/date/terms/compareTo/compareTo.md:34]:
```suggestion
The function throws a single exception, a [NullPointerException](https://www.codecademy.com/resources/docs/java/errors/nullpointerexception), if `argumentDate` is null.
```

**Review by caupolicandiaz — CHANGES_REQUESTED** [review]
@ChiragAgg5k I've completed my initial review. Please update the entry, and tag me when you've finished making changes. Thanks

**Comment by ChiragAgg5k** [comment]
@caupolicandiaz entry updated

**Comment by ChiragAgg5k** [comment]
Really sorry about the consecutive commits. Next time I will make sure to read more properly before pushing

**Comment by ChiragAgg5k** [comment]
Apologies for the consecutive commits. Next time I will try to read more carefully before pushing. Thanks

**Review by caupolicandiaz — APPROVED** [review]
@ChiragAgg5k @SSwiniarski LGTM

**Comment by SSwiniarski** [comment]
@ChiragAgg5k Congrats on your new entry! Live link here: https://www.codecademy.com/resources/docs/java/date/compareTo

image

You're currently anonymous. If you don't wish to be anonymous, check out this forum thread on how to link your GitHub and Codecademy profiles: https://discuss.codecademy.com/t/my-contribution-is-live-but-im-not-being-shown-as-a-contributor-why-is-this/758036  **Note:** it may take a few minutes for the change to register on the entry page.

**Pull request merged**